src/pages/index.tsx

```diff
@@ -12,63 +18,60 @@ const LOCATION = {
 const CENTER = [LOCATION.lat, LOCATION.lng];
 const DEFAULT_ZOOM = 3;
 
-// TODO type GraphQL Request!!
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const IndexPage = ({ data }: { data: any }) => {
-  const allProfiles = data.allMarkdownRemark.nodes;
-  const allProfilePics: { [index: string]: string } = {};
+const IndexPage = ({
+  data: { allMarkdownRemark, allImageSharp },
+}: {
+  data: FellowDataQuery;
+}) => {
+  const allProfiles = allMarkdownRemark.nodes;
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  data.allImageSharp.nodes.forEach((ele: any) => {
-    allProfilePics[ele.fluid.originalName] = ele.fluid.src;
-  });
+  const [isPortfolioModalOpen, setPortfolioModalOpen] = useState(false);
+  const [chosenFellow, setChosenFellow] = useState<Fellow | null>(null);
 
-  function mapEffect(baseMap: { leafletElement: L.Map } | null) {
-    if (!baseMap) return;
-    const { leafletElement } = baseMap;
+  // we likely don't want to generate this every render haha
+  const markers = useMemo(() => {
+    const ret: ReactElement[] = [];
 
     for (let i = 0; i < allProfiles.length; i++) {
-      const fellow = allProfiles[i].frontmatter;
-      const center = new L.LatLng(fellow.lat, fellow.long);
-      let social = '';
-      if (fellow.github) {
-        social += `<a href='https://github.com/${fellow.github}' target='_blank' rel="noreferrer"><i class="fab fa-github"></i></a>`;
-      }
-      if (fellow.linkedin) {
-        social += `<a href='https://www.linkedin.com/in/${fellow.linkedin}' target='_blank' rel="noreferrer"><i class="fab fa-linkedin"></i></a>`;
-      }
-      if (fellow.twitter) {
-        social += `<a href='https://twitter.com/${fellow.twitter}' target='_blank' rel="noreferrer"><i class="fab fa-twitter"></i></a>`;
-      }
+      const fellow = new Fellow(
+        allProfiles[i].frontmatter as FellowType,
+        allImageSharp,
+      );
 
-      const profilePop = `
-          <div class="profile">
-            <div><h3>${fellow.name}</h3></div>
-            <div><h4>${fellow.description}</h4></div>
-            <div class='divider'></div>
-            <div class='social-links'>
-              ${social}
-            </div>
-          </div>
-        `;
+      const center = new L.LatLng(fellow.lat, fellow.long);
 
-      L.marker(center, {
-        icon: L.icon({
-          className: 'icon',
-          iconUrl: `${allProfilePics[fellow.profilepic]}`,
-          iconSize: [50, 50],
-        }),
-      })
-        .addTo(leafletElement)
-        .bindPopup(profilePop);
+      ret.push(
+        <Marker
+          position={center}
+          key={fellow.name + fellow.lat}
+          icon={L.icon({
+            className: 'icon',
+            iconUrl: fellow.profilePictureUrl || 'none', // TODO handle missing img
```

Comment:
I think we can add a default img as a fallback. Alternatively, we can just hide the user.

Reply:
yes! Possibly a mlh logo. I'll try do it now before merging :+1: